internal/pkg/agent/cmd/run.go

```diff
@@ -374,3 +400,64 @@ func tryDelayEnroll(ctx context.Context, logger *logger.Logger, cfg *configurati
 	logger.Info("Successfully performed delayed enrollment of this Elastic Agent.")
 	return loadConfig(override)
 }
+
+func initTracer(agentName, version string, mcfg *monitoringCfg.MonitoringConfig) (*apm.Tracer, error) {
+	apm.DefaultTracer.Close()
```

Comment:
@AndersonQ I believe this should stay in the `init()` func, the tracing needs to be disabled as soon as we can and probably before the `initTracer()` method is called.

Reply:
orveral you said the PR is good, so I'm a wee unsure now, is this comment a blocker or not?

My main concern here is that the default tracer closing will be "hidden and lost" on the init function. Any future change might miss it as the whole code for initiating the tracer is here, on `initTracer`. Besides I hardly believe we'd send spans to a wrong server by closing the default tracer here instead of at the packages `init` func.
However if you say it's a blocker, I'm happy to change.

Reply:
I get the idea, let me approve this and lets try it out.